queries/java.scm

```diff
@@ -0,0 +1,330 @@
+    (#child-range! @switchStatementSubject 0 -1 true true)
+) @_.domain
+
+;;!! true ? 1 : 2
+(ternary_expression
+    condition: (_) @condition
+    consequence: (_) @branch
+) @condition.domain
+(ternary_expression
+    alternative: (_) @branch
+)
+
+;;!! true ? 1 : 2
+;;!  ^^^^^^^^^^^^
+(ternary_expression) @branch.iteration
+
+;;!! void myFunk(int value) {}
+;;!                  ^^^^^
+;;!  -------------------------
+(formal_parameters
+    (formal_parameter
+        (identifier) @name
+    ) @_.domain
+) @_.iteration
+
+;;!! int value = 0;
+;;!      ^^^^^
+;;!  --------------
+(_
+    (assignment_expression
+        left: (_) @name
+    ) @_.domain.start
+    ";"? @_.domain.end
+)
+
+;;!! Map<String, String>
+;;!     ^^^^^^^  ^^^^^^
+(type_arguments
+    (type_identifier) @type
+)
+
+;;!! List<String> list = value;
+;;!  ^^^^^^^^^^^^
+;;!  --------------------------
+(local_variable_declaration
+    type: (_) @type
+) @_.domain
+
+;;!! name = new ArrayList<String>();
+;;!             ^^^^^^^^^^^^^^^^^
+;;!         -----------------------
+(object_creation_expression
+    type: (_) @type
+) @_.domain
+
+;;!! name = new int[5];
+;;!             ^^^
+;;!         ----------
+(array_creation_expression
+    type: (_) @type
+) @_.domain
+
+;;!! void myFunk(int value) {}
+;;!              ^^^
+;;!              ---------
+(formal_parameter
+    type: (_) @type
+) @_.domain
+
+;;!! int size() {}
+;;!  ^^^
+;;!  -------------
+(method_declaration
+    type: (_) @type
+) @_.domain
+
+;;!! new test();
+;;!  ^^^^^^^^
+;;!  -----------
+(_
+    (object_creation_expression
+        (argument_list) @functionCallee.end.startOf
+    ) @functionCallee.start.startOf @_.domain.start
+    ";"? @_.domain.end
+)
+
+;;!! new test().bar();
+;;!  ^^^^^^^^^^^^^^
+;;!  -----------------
+(_
+    (method_invocation
+        (argument_list) @functionCallee.end.startOf
+    ) @functionCallee.start.startOf @_.domain.start
+    ";"? @_.domain.end
+)
+
+;;!! super();
+;;!  ^^^^^
+;;!  --------
+(explicit_constructor_invocation
+    (argument_list) @functionCallee.end.startOf
+) @functionCallee.start.startOf @_.domain
+
+;;!! for (int value : values) {}
+;;!                   ^^^^^^
+;;!  ---------------------------
+(enhanced_for_statement
+    type: (_) @type
+    name: (_) @name
+    value: (_) @value
+) @_.domain
+
+;;!! int value = 1;
+;;!              ^
+;;!           xxxx
+;;!  --------------
+(local_variable_declaration
+    (variable_declarator
+        name: (_) @name @value.removal.start.endOf
+        value: (_)? @value @value.removal.end
+    )
+) @_.domain
+(field_declaration
+    (variable_declarator
+        name: (_) @name @value.removal.start.endOf
+        value: (_)? @value @value.removal.end
+    )
+) @_.domain
+
+;;!! value = 1;
+;;!          ^
+;;!       xxxx
+;;!  ----------
+(_
+    (assignment_expression
+        left: (_) @value.removal.start.endOf
+        right: (_) @value @value.removal.end
+    ) @_.domain.start
+    ";"? @_.domain.end
+)
```

Comment:
I'd argue for using leading / trailing instead of removal for all of these

Reply:
If you prefer I could do that. Personally I don't think I have every use leading and trailing for anything else than white spaces, but I'm not against it.

Reply:
fixed

Reply:
Ha yeah was kind of thinking the same thing in the back of my head 😅. Better to be consistent tho I guess 🤷‍♂️